### PR TITLE
feat(command): add rule options

### PR DESCRIPTION
**Pull request by is-a-gamer**
now you can customize rules to uniformly filter messages

i added an ex06.py

### CODE REVIEW

**Review by TWT233 — CHANGES_REQUESTED** [review]
useful and practical feature, with minor design corner

File: khl/rule.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+class MessageRule(Rule):
+    """
+    can used in all message
+
+    @bot.command(name='hello', rule=MessageRule.at_me())
+    async def examples(msg: TextMsg, *args):
+        await msg.reply(f'hello')
+    """
+
+    def at_me(*args, **kwargs) -> (Callable[..., Coroutine]):
```
Comment (TWT233) [inline khl/rule.py:29]: first arg of method should be `self`

i think you mean
```python
@staticmethod
def at_me(*args, **kwargs):
```

Comment (TWT233) [reply khl/rule.py:29]: besides, these check functions is simple enough to be normal sync rather async, a bit misleading and unreadable

File: khl/rule.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from abc import ABC
+from khl import Msg
+
+
+class Rule(ABC):
```
[the remark this answers is not in the document]
Comment (TWT233) [reply khl/rule.py:6]: thx for ur reply and rev2, ya now i get what u wanna:

> ```python
>     def at_me(*args, **kwargs) -> (Callable[..., Coroutine]):
>         async def f(msg: Msg, *args) -> bool:
>             bot_id = await msg.ctx.bot.id()
>             return bot_id in msg.extra.get('mention')
> 
>         return f
> ```

you wanna a function that can be ref as `Rule.at_me()`, why not just use a package named as `Rule` and `def at_me(msg: Msg) -> bool:` in it, simpler and more intuitive.

no need to warp it with a bare `ABC`, the `Rule(ABC)` is not  interface just a container from ur description.

[the remark this answers is not in the document]
Comment (TWT233) [reply khl/rule.py:6]: btw, there is a huge refactor for this project that take place in branch:refactor/remake

in this turn khl.py will be granted a new structure, and get rid of some import circle, easier to extend

try moving this features there?

**Review by TWT233 — COMMENTED** [review]
rev2 comments, some talks in

File: khl/rule.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (is-a-gamer) [reply khl/rule.py:6]: ok,it will take some time

File: khl/rule.py (hunk 1 above)
Comment (is-a-gamer) [reply khl/rule.py:29]: if i have a function to wait,async function it works
or there are other solutions? i not familiar with py

Comment (TWT233) [reply khl/rule.py:29]: > if i have a function to wait,async function it works
> or there are other solutions? i not familiar with py

depends on the situation:

> assume the function to wait is `func()`

1. wait in another sync func: 
    - func() is sync: call it directly
    - func() is async: `asyncio.get_event_loop().run_until_complete(func())`
2. wait in another async func:
    - func() is sync: call it directly
    - func() is async: `await func()`

**Comment by is-a-gamer** [comment]
i create a new PR #20

**Pull request closed without merging**